### ISSUE TITLE
get_grad returns zero matrix with cpu

### ISSUE THREAD

**Issue by yttbgf** [PaddlePaddle/Paddle#11677]
use this code: http://www.paddlepaddle.org/docs/develop/book/02.recognize_digits/index.cn.html
def event_handler(event):
        if isinstance(event, paddle.event.EndIteration):
            if event.batch_id % 10 == 0:
                for p in parameters:
                    print "parameters:", parameters.get(p)
                    grad = parameters.get_grad(p)
                    print "gradients:", grad
                    print "gradients max abs:" max(grad.min(), grad.max(), key=abs)
            if event.batch_id % 100 == 0:
                print "Pass %d, Batch %d, Cost %f" % (
                    event.pass_id, event.batch_id, event.cost)
find grad matrix always zero!

**Comment by seiriosPlus** [comment]
能贴一下初始化parameters的代码么

**Comment by yttbgf** [comment]
@seiriosPlus 
  cost = paddle.layer.classification_cost(input=predict, label=label)

    parameters = paddle.parameters.create(cost)

    optimizer = paddle.optimizer.Momentum(
        learning_rate=0.1 / 128.0,
        momentum=0.9,
        regularization=paddle.optimizer.L2Regularization(rate=0.0005 * 128))

    trainer = paddle.trainer.SGD(
        cost=cost, parameters=parameters, update_equation=optimizer)

**Comment by yttbgf** [comment]
@seiriosPlus 
用的这个代码，只是paddle.init(use_gpu=False, trainer_count=1)
https://github.com/PaddlePaddle/book/blob/develop/02.recognize_digits/train.py

**Comment by seiriosPlus** [comment]
我执行了没有问题啊，你是用的最新版本的paddle么

**Comment by seiriosPlus** [comment]
参考文档：https://github.com/PaddlePaddle/Paddle/blob/d398b96823e6df38897fc881b5b71d7bc66013dc/doc/v2/faq/local/index_cn.rst#7--%E5%A6%82%E4%BD%95%E5%9C%A8%E8%AE%AD%E7%BB%83%E8%BF%87%E7%A8%8B%E4%B8%AD%E8%8E%B7%E5%BE%97%E5%8F%82%E6%95%B0%E7%9A%84%E6%9D%83%E9%87%8D%E5%92%8C%E6%A2%AF%E5%BA%A6
